core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2355,4 +2355,11 @@ package object config {
       .version("3.3.0")
       .intConf
       .createWithDefault(5)
+
+  private[spark] val ALLOW_CUSTOM_CLASSPATH_BY_PROXY_USER_IN_CLUSTER_MODE =
+    ConfigBuilder("spark.submit.proxyUser.allowCustomClasspathInClusterMode")
+      .internal()
+      .version("3.3.3")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
We should at least put this in the migration guide. The problem is that this is a breaking change so we didn't backport IIRC.

Reply:
@HyukjinKwon is this the one you're referring to: https://github.com/apache/spark/blob/master/docs/core-migration-guide.md?

- What if we default the setting `spark.submit.proxyUser.allowCustomClasspathInClusterMode` to true in 3.3.3? Then it won't be a breaking change anymore
- Shouldn't we also mention this property in the above migration guide from 3.3 to 3.4?

Reply:
I believe we can keep the existing behavior by having the default value `true` here. Then, it removes the breaking change issue and allows some users utilize this based on their situation. Could you change the default value, @degant ?

Reply:
Of course, the additional migration guide is always a welcome idea.

Reply:
Thanks @dongjoon-hyun I can default it to true here. And does `3.3.3` make sense? Is that the next release from branch-3.3?

Can you also share, what you're expecting in the migration guide? I can make the push the changes in the same PR. Only for 3.3 to 3.4? or also for 3.3?

Reply:
Set the default value to true and added a note in the migration guide for `3.3`. Please review when you get a chance @dongjoon-hyun and @HyukjinKwon, thanks!

I will raise another PR to add the migration guide for `3.4`